app/models/delivery_partner.rb

```diff
@@ -9,4 +9,8 @@ class DeliveryPartner < ApplicationRecord
   def declarations
     Declaration.for_delivery_partners(self)
   end
+
+  def cohorts_for_lead_provider(lead_provider)
+    cohorts.joins(:delivery_partnerships).where(delivery_partnerships: { lead_provider: lead_provider })
```

Comment:
If this is called on multiple lead providers, it will trigger multiple queries - I think this is better done in ruby

```ruby
delivery_partnerships
  .select { |partnership| partnership.lead_provider_id = lead_provider.id }
  .map(&:cohort)
```

this can then be called from elsewhere with

```ruby
DeliveryPartner
  .includes(delivery_partnerships: :cohorts)
  .map { |dp| dp.cohorts_for_lead_provider(currentlead_provider) }
```

Results in 3 queries versus uncontrolled number of queries